inst/rmarkdown/templates/jasa/resources/template.tex

```diff
@@ -0,0 +1,200 @@
+% $endif$
+% $if(mathfont)$
+%     \setmathfont(Digits,Latin,Greek)[$for(mathfontoptions)$$mathfontoptions$$sep$,$endfor$]{$mathfont$}
+% $endif$
+% $if(CJKmainfont)$
+%     \usepackage{xeCJK}
+%     \setCJKmainfont[$for(CJKoptions)$$CJKoptions$$sep$,$endfor$]{$CJKmainfont$}
+% $endif$
+
+$if(natbib)$
+\usepackage{natbib}
+$endif$
+$if(biblatex)$
+\usepackage[$if(biblio-style)$style=$biblio-style$,$endif$$for(biblatexoptions)$$biblatexoptions$$sep$,$endfor$]{biblatex}
+$for(bibliography)$
+\addbibresource{$bibliography$}
+$endfor$
+$endif$
+
+$if(listings)$
+\usepackage{listings}
+$endif$
+$if(lhs)$
+\lstnewenvironment{code}{\lstset{language=Haskell,basicstyle=\small\ttfamily}}{}
+$endif$
+$if(highlighting-macros)$
+$highlighting-macros$
+$endif$
+$if(verbatim-in-note)$
+\usepackage{fancyvrb}
+\VerbatimFootnotes % allows verbatim text in footnotes
+$endif$
+
+$for(header-includes)$
+$header-includes$
+$endfor$
+
+\begin{document}
+%% the square bracket argument will send term to running head in
+%% preprint, or running foot in reprint style.
+
+\title[$if(shorttitle)$$shorttitle$$endif$]{$title$}
+
+% ie
+%\title[JASA/Sample JASA Article]{Sample JASA Article}
+
+%% repeat as needed
+
+$for(author)$
+\author{$author.name$}
+% ie
+%\affiliation{Department1,  University1, City, State ZipCode, Country}
+\affiliation{$author.affiliation$}
+%% for corresponding author
+$if(author.email)$\email{$author.email$}$endif$
+%% for additional information
+$if(author.thanks)$\thanks{$author.thanks$}$endif$
+$endfor$
+
+% ie
+% \author{Author Four}
+% \email{author.four@university.edu}
+% \thanks{Also at Another University, City, State ZipCode, Country.}
+
+%% For preprint only,
+%  optional, if you want want this message to appear in upper left corner of title page
+% \preprint{}
+
+%ie
+%\preprint{Author, JASA}
+
+% optional, if desired:
+%\date{\today}
+
+$if(abstract)$
+\begin{abstract}
+% Put your abstract here. Abstracts are limited to 200 words for
+% regular articles and 100 words for Letters to the Editor. Please no
+% personal pronouns, also please do not use the words ``new'' and/or
+% ``novel'' in the abstract. An article usually includes an abstract, a
+% concise summary of the work covered at length in the main body of the
+% article.
+$abstract$
+\end{abstract}
+$endif$
+
+%% pacs numbers not used
+
+\maketitle
+
+%  End of title page for Preprint option --------------------------------- %
+
+%% See preprint.tex/.pdf or reprint.tex/.pdf for many examples
+
+$for(include-before)$
+$include-before$
+$endfor$
+
+%  Body of the article
+$body$
+
+%% before appendix (optional) and bibliography:
+% \begin{acknowledgments}
+%This research was supported by  ...
+% \end{acknowledgments}
+
+% -------------------------------------------------------------------------------------------------------------------
+%   Appendix  (optional)
+
+%\appendix
+%\section{Appendix title}
+
+%If only one appendix, please use
+%\appendix*
+%\section{Appendix title}
```

Comment:
I am trying to add an `appendices` option in the skeleton YAML which would list appendix files like:

```
---
...
appendices:
  - appendix-a.Rmd
  - appendix-b.Rmd
```

and I am trying to parse that and read it inside `jasa_article()` to render those files to `.tex` and input them in the template. I tried several ways, but no success. Is there a way of doing this? Or what could an alternative be?

Reply:
This would require to be handle in `jasa_article()` by modifying the pre_processor of its base format to render those child document, output them as tex and pass the content to Pandoc inside variable. 
It is not the easiest! 

In other template, this is often the simplest solution that is used: Pass it as raw tex in the pandoc variable like [header-includes]() :
* https://github.com/rstudio/rticles/blob/b0bbbc0a31c59952f4eab92e3df114cdc1aeed5b/inst/rmarkdown/templates/copernicus/skeleton/skeleton.Rmd#L73
* https://github.com/rstudio/rticles/blob/f0f03eb601d1c2e3bfe318dd0dccbdcda8896f86/inst/rmarkdown/templates/lipics/skeleton/skeleton.Rmd#L61

Not ideal but easy to maintain. 

One in-between way could be to process a tex file directly to include its content in the main document
```yaml
appendix: myappendix.tex
```
I think this would also require some pre processing in `jasa_article` to make it possible. 


So you may have been right in the first place... maybe the easier solution is to document it directly in the Rmd skeleton and not use in the template.


````markdown
---
<!-- yaml header --> 
---

<!-- Rmd content --> 

::: {.acknowledgments data-latex=""}
Thanks to...
:::

<!-- use \appendix* for only one appendix section -->
\appendix 

# Optional appendix A

Something 

# Optional appendix B

Something else

<!-- end of file --> 
````

About the fenced divs syntax: https://bookdown.org/yihui/rmarkdown-cookbook/custom-blocks.html

Maybe this easier as a user of this format. 
Sorry for having misleading you...

Reply:
No prob! I learnt a lot during the process. I will go with the custom blocks.

Reply:
FYI with next rmarkdown dev version we simplified it a little 

```markdown
::: {.acknowledgments latex=true}
Thanks to...
:::
```

Not to use here yet as it is still in dev version

Reply:
Done 7736eff, 20b2f9d, c4ee87b, fa7abd7.